sklego/meta/_grouped_utils.py

```diff
@@ -0,0 +1,62 @@
+from typing import Tuple
+
+import numpy as np
+import pandas as pd
+from scipy.sparse import issparse
+from sklearn.utils import check_array
+from sklearn.utils.validation import _ensure_no_complex_data
+
+from sklego.common import as_list
+
+
+def split_groups_and_values(
+    X, groups, name="", **kwargs
+) -> Tuple[pd.DataFrame, np.ndarray]:
+    general_checks(X, name=name)
+
+    try:
+        if isinstance(X, pd.DataFrame):
+            X_group = X.loc[:, as_list(groups)]
+            X_value = X.drop(columns=groups).values
+        else:
+            X_group = pd.DataFrame(X[:, as_list(groups)])
+            X_value = np.delete(X, as_list(groups), axis=1)
+    except (KeyError, IndexError):
+        raise ValueError(f"Could not drop groups {groups} from columns of X")
+
+    X_group = check_grouping_columns(X_group, **kwargs)
+    X_value = check_value_columns(X_value, **kwargs)
+
+    return X_group, X_value
+
+
+def general_checks(X, name):
```

Comment:
`_ensure_non_sparse` might be a better name.

Reply:
Renamed to `data_format_checks`, because it also checks non-complexity and shape (grouping should have at least two columns)